### DISCUSSION TITLE
Maintenance, scrapers and advertising?

### DISCUSSION THREAD

**Discussion by hans651** [hearchco/frontend#503] [Q&A]
hello, i accidentally stumbled up on this search engine on reddit and im curious will you be maintaining this? and does it scrape from google? will you be adveritising this?

**Comment by aleksasiriski** [comment]
Hello there! Which post on reddit brought you here, I'm curious?

Hearchco is very well in development and there is a bright future for this project to flower, so yeah maintaining in case any scraped engines break will happen.

I don't really use any social media except for Github so advertising will be left to the community :) And if you meant "will we have advertisements within results" the answer is a simple "No"

Hearchco scrapes a lot of engines (including google and bing), and you can find below the results listed all of the ones that returned that specific url. Better documentation is in the making.

**Comment by hans651** [comment]
thank you! i saw it on the reddit account of the 4get.co search engines dev guy! when i tested it hearch.co was just much faster than 4get/searxng its almost as fast as google for me! i wonder which country are you hosting it? is any telemetry data/fingerprinting happening on hearch?

**Comment by aleksasiriski** [comment]
> thank you! i saw it on the reddit account of the 4get.co search engines dev guy!

Great to hear :)

> when i tested it hearch.co was just much faster than 4get/searxng its almost as fast as google for me! i wonder which country are you hosting it?

Hearchco is hosted on almost every AWS region, so it's very close to every user on the planet (you can see exactly which regions [here]())

> is any telemetry data/fingerprinting happening on hearch?

No telemetry data / fingerprinting is being collected, there are logs that are associated with query letters (not the whole query, only each char from it) and query hashes (to separate each log line and see the trace of logs produces). Logging with that info is happening only when errors occured and the IP and any other info (useragent or whatever) is stripped from the log.

Until recently there was caching of results, but that was swiftly removed due to privacy concerns. I'm debating having another go at this that needs more discussion with the community.

**Reply by hans651** [reply]
thank you! appreciate the answers!
